authservice/auth_service.go

```diff
@@ -299,21 +308,254 @@ type RemoteAuthDataLoaderImpl struct {
+	return nil
+}
+
+// NewStaticTokenServiceAuth creates a new StaticTokenServiceAuth instance
+func NewStaticTokenServiceAuth(serviceToken string) (*StaticTokenServiceAuth, error) {
+	if serviceToken == "" {
+		return nil, fmt.Errorf("missing service token")
+	}
+	return &StaticTokenServiceAuth{ServiceToken: serviceToken}, nil
+}
+
+// AppOrgPair represents application organization pair access granted by a remote auth service
+type AppOrgPair struct {
+	AppID string
+	OrgID string
+}
+
+// Equals checks if two AppOrgPairs are equivalent
+func (ao AppOrgPair) Equals(other AppOrgPair) bool {
+	return ao.AppID == other.AppID && ao.OrgID == other.OrgID
+}
+
+// AccessToken represents an access token granted by a remote auth service
+type AccessToken struct {
+	Token     string `json:"access_token"`
+	TokenType string `json:"token_type"`
+}
+
+func (at AccessToken) String() string {
+	return fmt.Sprintf("%s %s", at.TokenType, at.Token)
+}
+
+type appOrgPairResponse struct {
+	AppID *string `json:"app_id"`
+	OrgID *string `json:"org_id"`
+}
+
+func appOrgPairFromResponse(res appOrgPairResponse) AppOrgPair {
+	return AppOrgPair{AppID: authutils.StringOrDefault(res.AppID, "all"), OrgID: authutils.StringOrDefault(res.OrgID, "all")}
+}
+
+type accessTokensResponse struct {
+	AppID       *string     `json:"app_id"`
+	OrgID       *string     `json:"org_id"`
+	AccessToken AccessToken `json:"token"`
+}
+
+// GetServiceAccountParams implements AuthDataLoader interface
+func (r *RemoteAuthDataLoaderImpl) GetServiceAccountParams() error {
+	if r.config == nil {
+		return fmt.Errorf("auth data loader is not configured")
+	}
+	req, err := r.config.BuildServiceAccountParamsRequest()
+	if err != nil {
+		return fmt.Errorf("error creating service account params request: %v", err)
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("error requesting service account params: %v", err)
+	}
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("error reading body of service account params response: %v", err)
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("error getting service account params: %d - %s", resp.StatusCode, string(body))
+	}
+
+	var paramsResponse []appOrgPairResponse
+	err = json.Unmarshal(body, &paramsResponse)
+	if err != nil {
+		return fmt.Errorf("error on unmarshal service account params response: %v", err)
+	}
+
+	r.updateCachedPairs(paramsResponse)
+
+	return nil
+}
+
 // GetAccessToken implements AuthDataLoader interface
-func (r *RemoteAuthDataLoaderImpl) GetAccessToken() error {
+func (r *RemoteAuthDataLoaderImpl) GetAccessToken(appID string, orgID string) error {
+	if r.config == nil {
+		return fmt.Errorf("auth data loader is not configured")
+	}
+	req, err := r.config.BuildAccessTokenRequest(authutils.StringOrNil(appID, "all"), authutils.StringOrNil(orgID, "all"))
```

Comment:
We should probably make sure that all of our APIs are aware of the switch to "all" instead of nil/"". If we fix this, there should be no need for this type of conversion.